### PR TITLE
feat: introduce monitoring for CheckSecrets

### PR DESCRIPTION
**Pull request by smartcontracts**
Introduces a new monitoring package that keeps track of CheckSecrets contracts used by a particular Drippie contract. Queries the Drippie contract for new drips and stores those drips for later. Iterates over stored drips during each loop to determine if the provided secrets are revealed or not.

### CODE REVIEW

File: op-monitorism/secrets/monitor.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+	}
+
+	// use this to filter for DripCreated events query
+	m.drippie.DrippieFilterer.FilterDripCreated(
+		&bind.FilterOpts{
+			Start: fromBlockNumber,
+			End:   &toBlockNumber,
+		},
+		nil,
+	)
```
Comment (ajsutton) [inline op-monitorism/secrets/monitor.go:145]: Does this actually do anything? We aren't using any return value.

**Review by ajsutton — APPROVED** [review]
LGTM.  Just confirming my understanding that we need to scan the full block range on each startup to find all drips.  That will get slower and slower over time but it likely won't be a problem for a fair while.

We should also expose a metric that reports the number of known drips so that we can sanity check on that and alert if it ever goes down (which indicates we missed a drip somehow).  I've seen no end of problems with execution clients failing to return all logs so we should make sure we have some kind of check around that.

File: op-monitorism/secrets/monitor.go (hunk 1 above)
Comment (smartcontracts) [reply op-monitorism/secrets/monitor.go]: Oops, leftover code

**Comment by smartcontracts** [comment]
Actually @ajsutton I've decided to open up a PR here to add an array into `Drippie` so that using the event query isn't necessary. I'd rather not use an event query since it's relatively slow and the number of elements inside of Drippie is going to be limited enough where iterating over an array is perfectly fine.

**Comment by raffaele-oplabs** [comment]
The code needs to be integrated with monitorism, so it should be also modified here: op-monitorism/cmd/monitorism/cli.go
Also we need a section in the readme.

**Comment by smartcontracts** [comment]
> The code needs to be integrated with monitorism, so it should be also modified here: op-monitorism/cmd/monitorism/cli.go Also we need a section in the readme.

This is done now.

**Review by raffaele-oplabs — APPROVED** [review]

**Pull request merged**